distribution/tools/plugin-cli/src/test/java/org/opensearch/plugins/InstallPluginCommandTests.java

```diff
@@ -256,29 +256,52 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
     }
 
     /** creates a plugin .zip and returns the url for testing */
-    static String createPluginUrl(String name, Path structure, String... additionalProps) throws IOException {
-        return createPlugin(name, structure, additionalProps).toUri().toURL().toString();
-    }
-
-    static void writePlugin(String name, Path structure, String... additionalProps) throws IOException {
-        String[] properties = Stream.concat(
-            Stream.of(
-                "description",
-                "fake desc",
-                "name",
-                name,
-                "version",
-                "1.0",
-                "opensearch.version",
-                Version.CURRENT.toString(),
-                "java.version",
-                System.getProperty("java.specification.version"),
-                "classname",
-                "FakePlugin"
-            ),
-            Arrays.stream(additionalProps)
-        ).toArray(String[]::new);
-        PluginTestUtil.writePluginProperties(structure, properties);
+    static String createPluginUrl(String name, boolean withCustomFolder, Path structure, String... additionalProps) throws IOException {
+        return createPlugin(name, withCustomFolder, structure, additionalProps).toUri().toURL().toString();
+    }
+
+    static void writePlugin(String name, boolean withCustomFolder, Path structure, String... additionalProps) throws IOException {
+        if (withCustomFolder) {
+            String[] properties = Stream.concat(
+                Stream.of(
+                    "description",
+                    "fake desc",
+                    "name",
+                    name,
+                    "version",
+                    "1.0",
+                    "opensearch.version",
+                    Version.CURRENT.toString(),
+                    "java.version",
+                    System.getProperty("java.specification.version"),
+                    "classname",
+                    "FakePlugin",
+                    "folderName",
+                    "customFolder"
```

Comment:
Since it's a folder, let's change the example to `custom-folder`, we don't want to encourage camelCase folder names.

Reply:
Thats true! Updated all the custom folder names to kebab-case.